editor/src/components/canvas/canvas-strategies/strategies/convert-to-absolute-and-move-strategy.tsx

```diff
@@ -658,46 +701,17 @@ function createSetParentsToFixedSizeCommands(
       return []
     }
     return firstAncestorsHonoringPropsSize.flatMap((ancestor) => {
-      const parentMetadata = MetadataUtils.findElementByElementPath(metadata, ancestor)
-      const parentLocalFrame = parentMetadata?.localFrame
-      if (parentLocalFrame == null || !isFiniteRectangle(parentLocalFrame)) {
-        return []
-      }
-
       const parentElement = MetadataUtils.getJSXElementFromMetadata(metadata, ancestor)
       if (parentElement == null) {
         return []
       }
 
-      const setWidthCommands = isCollapsingParent(parentElement, 'width')
-        ? [
-            setCssLengthProperty(
-              'always',
-              ancestor,
-              PP.create('style', 'width'),
-              setExplicitCssValue(cssPixelLength(parentLocalFrame.width)),
-              null,
-            ),
-          ]
-        : []
-      const setHeightCommands = isCollapsingParent(parentElement, 'width')
-        ? [
-            setCssLengthProperty(
-              'always',
-              ancestor,
-              PP.create('style', 'height'),
-              setExplicitCssValue(cssPixelLength(parentLocalFrame.height)),
-              null,
-            ),
-          ]
-        : []
-
-      const commands = [...setWidthCommands, ...setHeightCommands]
-      if (commands.length === 0) {
+      if (!isHuggingParent(parentElement, 'width') && !isHuggingParent(parentElement, 'height')) {
         return []
       }
+
       return [
-        ...commands,
+        ...sizeToVisualDimensions(metadata, pathTree, ancestor),
         showToastCommand('Parent is set to fixed size', 'NOTICE', 'set-parent-to-fixed-size'),
```

Comment:
I suspect we don't need this toast any more if we're providing this as an explicit option